baseapp/protocompat/protocompat.go

```diff
@@ -0,0 +1,191 @@
+package protocompat
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	gogoproto "github.com/cosmos/gogoproto/proto"
+	"google.golang.org/grpc"
+	proto2 "google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+)
+
+var (
+	gogoType    = reflect.TypeOf((*gogoproto.Message)(nil)).Elem()
+	protov2Type = reflect.TypeOf((*proto2.Message)(nil)).Elem()
+)
+
+type Handler = func(ctx context.Context, message gogoproto.Message) (gogoproto.Message, error)
+
+func MakeHybridHandler(cdc codec.BinaryCodec, sd *grpc.ServiceDesc, method grpc.MethodDesc, handler interface{}) (Handler, error) {
+	methodFullName := protoreflect.FullName(fmt.Sprintf("%s.%s", sd.ServiceName, method.MethodName))
+	desc, err := gogoproto.HybridResolver.FindDescriptorByName(methodFullName)
+	if err != nil {
+		return nil, err
+	}
+	methodDesc, ok := desc.(protoreflect.MethodDescriptor)
+	if !ok {
+		return nil, fmt.Errorf("invalid method descriptor %s", methodFullName)
+	}
+
+	funcType, err := getFuncType(handler, method.MethodName)
```

Comment:
I don't think we should depend on these internal details. `MethodDesc` has the thing we should call and we can introspect that to get the type